lib/olly_trace/olly_format_backend/olly_format_fuchsia.ml

```diff
@@ -1,41 +1,56 @@
 open Olly_format_backend
-open Tracing
 
 let name = "fuchsia"
 let description = "Perfetto"
 
-type trace = { doms : Trace.Thread.t array; file : Trace.t }
+module Trace = Trace_fuchsia.Writer
+
+type trace = { 
+  doms : Trace.Thread_ref.t array; 
+  buf : Trace_fuchsia.Buf_chain.t;
+  subscriber : Trace_fuchsia.Subscriber.t;
+  exporter : Trace_fuchsia.Exporter.t;
+}
+
+let flush trace =
+  Trace_fuchsia.Buf_chain.ready_all_non_empty trace.buf;
+  Trace_fuchsia.Buf_chain.pop_ready trace.buf ~f:trace.exporter.write_bufs; 
+  trace.exporter.flush ()
 
 let create ~filename =
-  let file = Trace.create_for_file ~base_time:None ~filename in
+  let buf_pool = Trace_fuchsia.Buf_pool.create () in
+  let buf = Trace_fuchsia.Buf_chain.create ~sharded:false ~buf_pool () in
+  let oc = Out_channel.open_bin filename in
+  let exporter = Trace_fuchsia.Exporter.of_out_channel ~close_channel:true oc in 
+  let subscriber = Trace_fuchsia.Subscriber.create ~buf_pool ~pid:0 ~exporter () in
+  (* Adds the headers to output *)
+  Trace_fuchsia.Subscriber.Callbacks.on_init subscriber ~time_ns:0L;
   let doms =
     let max_doms = 128 in
     Array.init max_doms (fun i ->
         (* Use a different pid for each domain *)
-        Trace.allocate_thread file ~pid:i ~name:(Printf.sprintf "Ring_id %d" i))
+        Trace.Thread_ref.ref (i + 1))
   in
-  { doms; file }
+  { doms; buf; subscriber; exporter }
 
-let close trace = Trace.close trace.file
-let ts_to_span ts = ts |> Int64.to_int |> Core.Time_ns.Span.of_int_ns
+let close trace =
+  flush trace;
+  Trace_fuchsia.Subscriber.close trace.subscriber 
 
 let emit trace evt =
   let open Event in
-  let thread = trace.doms.(evt.ring_id)
-  and category = "PERF"
-  and time = ts_to_span evt.ts
+  let t_ref = trace.doms.(evt.ring_id)
+  and time_ns = evt.ts
   and name = evt.name in
   match evt.kind with
   | SpanBegin | SpanEnd ->
       let write =
-        if evt.kind = SpanBegin then Trace.write_duration_begin
-        else Trace.write_duration_end
+        if evt.kind = SpanBegin then Trace.Event.Duration_begin.encode
+        else Trace.Event.Duration_end.encode
```

Comment:
you could direct use the `Trace_fuchsia.Subscriber' callbacks here, too, without going through `Trace.Event`.